CHANGELOG.md

```diff
@@ -1,5 +1,24 @@
 # beeline-go changelog
 
+## 1.12.0 2023-06-05
+
+### ⚠️ Breaking Changes ⚠️
+
+Minimum Go version required is 1.17
```

Comment:
Just realizing [the docs](https://docs.honeycomb.io/getting-data-in/beeline/go/#requirements) still show a minimum Go version of 1.11, can we update those when this release is ready too?

Reply:
Created docs PR to update min version:
- https://github.com/honeycombio/docs/pull/1784